### ISSUE TITLE
'%' character in elfeed-search-mode header-line-format 

### ISSUE THREAD

**Issue by gusbrs** [skeeto/elfeed#415]
Hi skeeto,

this is a small one, but easy to fix too.

When `elfeed-search-filter` contains the `%` character, the header in `elfeed-search-mode` garbles the character.  What happens is that `elfeed-search-filter` is passed directly to `header-line-format` by `elfeed-search--header`, and the `%` will then be interpreted as a "%-Construct" by the header.

As far as I can tell, just passing `(replace-regexp-in-string "%" "%%" elfeed-search-filter)` instead of `elfeed-search-filter` in `elfeed-search--header` should be enough.

**Comment by skeeto** [comment]
Thanks for pointing this out! I had no idea it did interpolation like 
that. It should now be fixed by 7186995. Rather than escape %-constructs 
as you had suggested, I figured out how to disable interpolation.

**Comment by gusbrs** [comment]
Wow, that was fast! Thank you!

**Comment by gusbrs** [comment]
Skeeto, test report: I might be missing some piece, but I tested the commit by redefining `header-line-format` in `elfeed-search-mode-hook` just in my init file. The search-filter string is indeed correct even in the presence of a "%", but the text-properties of the whole header (colors) seem to be gone. If that's not happening on your side, great, it might be something in the procedure I used to do it. But I thought worth it to report anyway.
